### ISSUE TITLE
Local Health Check Warning reasons

### ISSUE THREAD

**Issue by leshadubenyuk** [dotnet/orleans#7065]
Hello,
after update to .Net 5 my Orleans project started to have warnings about GC timer is not triggered at time. It happens regularly, for already quite some time, and i am not sure where to look for a reason of this. 
Logs:
`Orleans.Runtime.MembershipService.LocalSiloHealthMonitor - Health check participant Orleans.Runtime.Catalog is reporting that it is unhealthy with complaint: Catalog.GCTimer timer should have fired at 3/26/2021 6:27:55 AM, which is 00:00:07.2431620 ago`
`Orleans.Runtime.MembershipService.LocalSiloHealthMonitor - Self-monitoring determined that local health is degraded. Degradation score is 1/8 (lower is better). Complaints: Health check participant Orleans.Runtime.Catalog is reporting that it is unhealthy with complaint: Catalog.GCTimer timer should have fired at 3/26/2021 6:27:55 AM, which is 00:00:07.2431620 ago`

I would really appreciate if someone can advise on how to tackle this issue. I guess its not very serious since no issues with application's operation, but it surely doesn't look like indented.

**Comment by ReubenBond** [comment]
If the application is operating without issue, you can consider ignoring it.

However, it does typically indicate a low severity issue. For example, it might indicate a long garbage collection pause. The easiest way to find out is by profiling your application underload, unfortunately (since that is not a trivial exercise).

Do you see any other warnings in your application logs?

**Comment by leshadubenyuk** [comment]
No, nothing that could be related. I would expect to see logs more from AsyncTimer, for example on next tick, but i don't see any, which makes me feel suspicious. 
[https://github.com/dotnet/orleans/blob/10af0f4af588cd4aa45cb3e250dfbffa389d59c7/src/Orleans.Runtime/Timers/AsyncTimer.cs#L87, AsyncTimer.cs].

**Comment by shersh** [comment]
I have a lot of warnings too.

**Comment by ReubenBond** [comment]
If you see a lot of warnings, then it's worth investigating by capturing logs and potentially CPU traces. It's fairly common for some thread pool misuse to cause starvation. Eg, sync over async / async over sync. Very high logging volume with the console logger can also cause issues. It's best to examine these performance issues individually, and we can help: send me a message in Gitter and I can try to troubleshoot with you.
